### ISSUE TITLE
Does Concatenate order matters?

### ISSUE THREAD

**Issue by ecilay** [facebookincubator/AITemplate#995]
I have below code, which concats the input tensor before conv layer, assume the input tensor shape is (N, H, W, Cin).
This is the original corresponding Pytorch code:
`x = torch.nn.functional.pad(x, pad, mode="constant", value=0)`.

Conversion code 1:
`
T1 = ops.full()(shape=[N, 1, W, Cin], fill_value=0)
R = ops.concatenate()(inputs=[x, T1], dim=1)
T2 = ops.full()(shape=[N, H + 1, 1, Cin], fill_value=0)
x = ops.concatenate()(inputs=[R, T2], dim=2)
`,
and this will give different results than
Conversion code 2:
`
T1 = ops.full()(shape=[N, H, 1, Cin], fill_value=0)
R = ops.concatenate()(inputs=[x, T1], dim=2)
T2 = ops.full()(shape=[N, 1, W+1, Cin], fill_value=0)
x = ops.concatenate()(inputs=[R, T2], dim=1)
`.

This code is run many times during inference. I find when I use the bottom block of code, the conversion results are bad when W is close to 2 times of H. For square tensor, the conversion works fine. Only the 1st conversion gives correct results regardless of dimensions.

**Comment by kadeng** [comment]
thanks for your report, we will try to reproduce the issue and then get back to you.
